### ISSUE TITLE
libint: enable cmake build

### ISSUE THREAD

**Issue by sheepforce** [NixOS/nixpkgs#132879]
### Describe the bug
`libint` is a two step procedure build: the libint code generator creates an archive of the actual libint C++ code, which is then compiled. The libint code-generator uses autotools-style build, while the subsequent C++ compilation can be done on in the same step using autotools-style build, or alternatively using a CMake+Ninja-style build. Currently the libint derivation performs the code generation step and the actual compilation as a single derivation. While this is working just fine, the resulting derivation cannot be found by CMake and the documentation actually discourages using autotools for the code compilation step, see https://github.com/evaleev/libint/wiki#compiling-libint-library.

E.g. [Psi4 1.4](https://github.com/psi4/psi4) recently switched from libint1 to libint2 and cannot find libint2 with its CMake based build system (as the libint cmake files are actually missing).

A possible solution could be to split libint into the two derivations. This would also reduce the enormous build time of this single derivation. The downside is, that the code-generation produces just a source-code tarball, which is a little awkward as a nix derivation.

### Steps To Reproduce
1. `nix-build -A libint`
2. `results/share/cmake` is missing

### Expected behavior
Perform the code-compilation step with cmake as suggested in the documentation.

### Notify maintainers
@markuskowa 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.45, NixOS, 21.05.2075.ff1ea3a36c1 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210707_02dd6bb`
 - channels(root): `"nixos-21.05.2075.ff1ea3a36c1"`
 - channels(phillip): `"nixpkgs-21.11pre300826.f51caec3fc1, qchem"`
 - nixpkgs: `/home/phillip/.nix-defexpr/channels/nixpkgs`
```

**Comment by markuskowa** [comment]
We could just add a cmake file to the output (e.g. into `$out/share/cmake`). That is a common practice, for example, for packages that are lacking a `pkg-config` file.

**Comment by sheepforce** [comment]
Also sounds perfectly reasonable. Building with CMake could still be wort a try, though. I guess there is a good chances that libint 2.7 onwards could remove autotools-based builds for the second step, as they already declared it deprecated.

**Comment by sheepforce** [comment]
Closed by #133258
